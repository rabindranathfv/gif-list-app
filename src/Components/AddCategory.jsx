import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = (props) => {

    const { setCategory } = props;
    const [inputValue, setInputValue] = useState('');

    const addHandler = (e) => {
        setInputValue( e.target.value );
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        if ( inputValue.trim().length > 2) {
            setCategory(cat => [ inputValue]);
            console.log('AQUI');
            setInputValue('');
            console.log('AQUI 4');
        }
    }

    return (
        <form onSubmit={handleSubmit}>
            { inputValue && <p>{ inputValue }</p> }
            <input type="text" value={inputValue} onChange={addHandler} />
        </form>
    )
}

AddCategory.propTypes = {
    setCategory: PropTypes.func.isRequired
}