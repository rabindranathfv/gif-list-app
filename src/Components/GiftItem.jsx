import React from 'react'

export const GiftItem = ({ gift }) => {

    return (
        <div key={gift.id} className='card'>
            <p > {gift.title}</p>  
            <img src={gift.url} alt={gift.title} />
        </div>
    )
}
