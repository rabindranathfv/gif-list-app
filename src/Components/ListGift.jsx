import React, { useState } from 'react';

export const ListGift = (props) => {

    const { gifts } = props;
    const [categories, setCategories] = useState(gifts);
    return (
        <div>
            <ul>
            {
                categories.map( cat => {
                    return <li key={cat.id}> {cat.name} </li>
                })
            }
            </ul>
        </div>
    )
}
