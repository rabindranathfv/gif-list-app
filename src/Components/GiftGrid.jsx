import React, { useEffect, useState } from 'react'
import PropTypes, { string } from 'prop-types';

import { GiftItem } from './GiftItem';

const API_URL = `https://api.giphy.com/v1/gifs/search`;
const API_KEY = `6ijpJmhvX9OshRF2lAHuBYHojzB9bZjX`;

export const GiftGrid = props => {

    const { category, searchCategory } = props;

    const [gifts, setGifts] = useState([]);

    const getGifts = async() => {
        const urlQuery = `${API_URL}?q=${searchCategory}&limit=10&api_key=${API_KEY}`
        const resp = await fetch( urlQuery);
        const { data } = await resp.json();
        
        const giftItem = data.map( gif => {
            return { 
                id: gif.id,
                title: gif.title,
                url:  gif.images?.downsized_medium.url
            }
        })
        setGifts( giftItem);
    }

    useEffect(() => {
        getGifts();
    }, []);

    return (
        <div className='card-grid'>
            <ul>
            {
                gifts.map( (gif) => {
                    return <GiftItem gift={gif} /> 
                })
            }
            </ul>
        </div>
    )
}

GiftGrid.propTypes = {
    category: PropTypes.shape({
        id: string,
        title: string,
        url: string
      }).isRequired,
}
