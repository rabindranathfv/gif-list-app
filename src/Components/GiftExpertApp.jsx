import React, {useState} from 'react';
import { AddCategory } from './AddCategory';
import { GiftGrid } from './GiftGrid';

const GiftExpertApp = () => {

    const [category, setCategory] = useState('');

    console.log('CATEGORY IN MAIN COMPONENT', category);

    return (
        <div>
            <h2> GifExpertAPP </h2>
            <AddCategory setCategory={setCategory} />
            <hr />
            <GiftGrid category={category}  />
        </div>
    )
}

GiftExpertApp.defaultProps = {
    category: 'dragon ball',
}

export default GiftExpertApp
