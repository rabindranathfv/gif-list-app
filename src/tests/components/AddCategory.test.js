import { shallow } from 'enzyme';
import React from 'react';
import '@testing-library/jest-dom';

import { AddCategory } from '../../Components/AddCategory';

describe('AddCategory component', () => {
    const setCategoryMockResp = '';
    const setCategoryMock = jest.fn(() => Promise.resolve(setCategoryMockResp));
    let wrapper =  shallow( <AddCategory setCategory={setCategoryMock} />);

    beforeEach( () => {
        jest.clearAllMocks();
        wrapper = shallow( <AddCategory setCategory={setCategoryMock} />);
    });

    test('should create component', () => {
        wrapper.debug();
        expect( wrapper).toMatchSnapshot();
    });

    test('should trigger addHandler', () => {
        const input = wrapper.find('input');
        const inputText = 'input test value';
        input.simulate('change', { 
            target: { value: inputText }
            }
        );
        const textSearch = wrapper.find('p');
        expect(textSearch.text().trim()).toBe(inputText);
    });
    
    test('should call handleSubmit but do not call setCategory', () => {
        const input = wrapper.find('form');
        input.simulate('click', { preventDefault: () => {} });
        expect(setCategoryMock).not.toBeCalled();
    });
    
    
    
})
