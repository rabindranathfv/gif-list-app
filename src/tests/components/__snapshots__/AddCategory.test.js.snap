// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCategory component should create component 1`] = `
<form
  onSubmit={[Function]}
>
  <p>
    search by category
  </p>
  <input
    onChange={[Function]}
    type="text"
    value="search by category"
  />
</form>
`;
